src/pretix/base/services/cross_selling.py

```diff
@@ -0,0 +1,201 @@
+        if category.cross_selling_condition == 'discounts':
+            my_item_pks = [item.id for item in category.items.all()]
+            potential_discount_items = {
+                item.pk: (max_count, discount_rule)
+                for item, max_count, discount_rule in self._potential_discounts_by_item_for_current_cart
+                if max_count > 0 and item.pk in my_item_pks and item.is_available()
+            }
+
+            return category.items.filter(pk__in=potential_discount_items), potential_discount_items
+
+    @cached_property
+    def _potential_discounts_by_item_for_current_cart(self):
+        potential_discounts_by_cartpos = defaultdict(list)
+
+        from ..services.pricing import apply_discounts
+        self._discounted_prices = apply_discounts(
+            self.event,
+            self.sales_channel,
+            [
+                (cp.item_id, cp.subevent_id, cp.line_price_gross, bool(cp.addon_to), cp.is_bundled,
+                 cp.listed_price - cp.price_after_voucher)
+                for cp in self.cartpositions
+            ],
+            collect_potential_discounts=potential_discounts_by_cartpos
+        )
+
+        # flatten potential_discounts_by_cartpos (a dict of lists of potential discounts) into a set of potential discounts
+        # (which is technically stored as a dict, but we use it as an OrderedSet here)
+        potential_discount_set = dict.fromkeys(
+            info for lst in potential_discounts_by_cartpos.values() for info in lst)
+
+        # sum up the max_counts and pass them on (also pass on the discount_rules so we can calculate actual discounted prices later):
+        # group by benefit product
+        # - max_count for product: sum up max_counts
+        # - discount_rule for product: take first discount_rule
+
+        def discount_info(item, infos_for_item):
+            infos_for_item = list(infos_for_item)
+            return (
+                item,
+                sum(max_count for (item, discount_rule, max_count, i) in infos_for_item),
+                next(discount_rule for (item, discount_rule, max_count, i) in infos_for_item)
+            )
+
+        return [
+            discount_info(item, infos_for_item) for item, infos_for_item in
+            groupby(
+                sorted(
+                    (
+                        (item, discount_rule, max_count, i)
+                        for (discount_rule, max_count, i) in potential_discount_set.keys()
+                        for item in discount_rule.benefit_limit_products.all()
+                    ),
+                    key=lambda tup: tup[0].pk
+                ),
+                lambda tup: tup[0])
+        ]
+
+    def _prepare_items(self, subevent, items_qs, discount_info):
+        items, _btn = get_grouped_items(
+            self.event,
+            subevent=subevent,
+            voucher=None,
+            channel=self.sales_channel,
+            base_qs=items_qs,
+            allow_addons=True,
```

Comment:
Really?

Reply:
Probably not... (shouldn't make a difference as it's mutually exclusive to appearing in cross-selling queryset, but should still be False)
```suggestion
            allow_addons=False,
```